### DISCUSSION TITLE
[Help] Git LFS file version issue

### DISCUSSION THREAD

**Discussion by Cactusinhand** [git-lfs/git-lfs#4809] [Q&A]
Hi @bk2204 
If I migrate a file into Git-LFS, it becomes LFS pointer file, what I edit then is just LFS pointer info, not their original content, then how can I version it? should I export it to normal file first?

**Comment by bk2204** [comment]
If you've run `git lfs install` on your system at least once, then when you checkout out a repository with LFS files, they should end up being checked out with their original content, not the pointer file.  If you haven't done that, you should do so, which will modify your `~/.gitconfig` file to set the appropriate values for this to work.

Once you've done that, if you still have repositories missing their original content, you can run `git lfs pull`, which should cause the original content to be checked out.

**Reply by Cactusinhand** [reply]
OK, my understanding is:
a).`git lfs install` will install some hooks in `.git/hooks/`, such as `pre-push`, `post-checkout`, which will be called when do git push or checkout action. And it will also configure `~/.gitconfig` file with clean/smudge filters.
b). `git lfs pull` will checkout blob original content from `.git/lfs/objects/`.

So,  what I should do after I converted some normal files into LFS pointer file is to run `git lfs pull`, then I can edit them and version them normally.

**Reply by bk2204** [reply]
Both a and b are correct.  You shouldn't need to run `git lfs pull` in normal operation, since the files should automatically be checked out by the smudge and clean filters, and adding a new file shouldn't cause the contents in your working tree to be changed.
